ui_testing/testcases/basic_tests/test05_contacts.py

```diff
@@ -426,3 +429,233 @@ def test014_edit_approach_overview_button(self):
+            self.assertEqual(row_data['Country'], data_to_filter_with)
+            counter = counter +1
+    
+    def test025_filter_by_contact_type(self):
+        """
+        Contacts: Filter Approach: Make sure you can filter by type
+        LIMS-6421
+        """
+        
+        data_to_filter_with = self.contacts_api.get_first_record_with_data_in_attribute(attribute='type')
+        self.assertNotEqual(data_to_filter_with, False)
+        data_to_filter_with = self.contact_page.get_mapped_contact_type(contact_type=data_to_filter_with[0])
+        
+        self.base_selenium.LOGGER.info('filter with {}'.format(data_to_filter_with))
+        self.contact_page.apply_filter_scenario(filter_element='contact:type_filter', filter_text=data_to_filter_with)
+        table_records = self.contact_page.result_table()
+        counter = 0
+        while counter < (len(table_records) -1):
+            row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=table_records[counter])
+            contact_type = row_data['Type'].split(', ')
+            self.assertIn(data_to_filter_with, contact_type)
+            counter = counter +1
+   
+    def test026_filter_by_contact_phone(self):
+        """
+        Contacts: Filter Approach: Make sure you can filter by phone
+        LIMS-6412
+        """
+        
+        data_to_filter_with = self.contacts_api.get_first_record_with_data_in_attribute(attribute='phone')
+        self.assertNotEqual(data_to_filter_with, False)
+        self.base_selenium.LOGGER.info('filter with {}'.format(data_to_filter_with))
+        self.contact_page.apply_filter_scenario(filter_element='contact:phone', filter_text=data_to_filter_with, field_type='text')
+        table_records = self.contact_page.result_table()
+        counter = 0
+        while counter < (len(table_records) -1):
+            row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=table_records[counter])
+            self.assertEqual(row_data['Phone'], data_to_filter_with)
+            counter = counter +1
+    
+    def test027_filter_by_contact_location(self):
+        """
+        Contacts: Filter Approach: Make sure you can filter by location
+        LIMS-6417
+        """
+        
+        data_to_filter_with = self.contacts_api.get_first_record_with_data_in_attribute(attribute='location')
+        self.assertNotEqual(data_to_filter_with, False)
+        self.base_selenium.LOGGER.info('filter with {}'.format(data_to_filter_with))
+        self.contact_page.apply_filter_scenario(filter_element='contact:location', filter_text=data_to_filter_with, field_type='text')
+        table_records = self.contact_page.result_table()
+        counter = 0
+        while counter < (len(table_records) -1):
+            row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=table_records[counter])
+            self.assertEqual(row_data['Location'], data_to_filter_with)
+            counter = counter +1
```

Comment:
parameterized

Reply:
what i got from diana, that you wanted each one in a specific testcase for it, not as a parameters

Reply:
@mazenmmbadr I'm not sure if she got my point clear or not but those cases are identical with different data > parameterized is the best here

Reply:
but it is implemented in articles and other testcases using the same way, also not all the data are handled the same, so using parameters will be just long test case with multiple if conditions to handle the data that needs special handling